### DISCUSSION TITLE
What is the recommended memory limiter/ballast config following the ballast deprecation?

### DISCUSSION THREAD

**Discussion by bonnici** [open-telemetry/opentelemetry-collector#9264] [Q&A] [answered]
The [Memory Limiter Processor README file](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiterprocessor/README.md) still says:
> It is highly recommended to configure ballastextension as well as the memory_limiter processor on every collector. The ballast should be configured to be 1/3 to 1/2 of the memory allocated to the collector. The memory_limiter processor should be the first processor defined in the pipeline (immediately after the receivers). This is to ensure that backpressure can be sent to applicable receivers and minimize the likelihood of dropped data when the memory_limiter gets triggered.

The [v0.92 changelog](https://github.com/open-telemetry/opentelemetry-collector/releases/tag/v0.92.0) lists the ballast extension as deprecated. What is the new recommended config to use? Should we set GOMEMLIMIT to be what the old ballast size was?

**Comment by atoulme** [comment, marked as answer]
Feel free to review and accept or provide comments to this PR as it touches that file: https://github.com/open-telemetry/opentelemetry-collector/pull/9152

**Reply by bonnici** [reply]
Thank you! That made it more clear - I've set memory limit to about 90% of the available memory, spike limit to about 20% of that, and GOMEMLIMIT to about 20% lower than the 90% hard limit. I don't think I should accept the PR though since I'm pretty clueless about Go and how it uses memory.

**Reply by atoulme** [reply]
No worries. Happy that helped.
